tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test.py

```diff
@@ -1024,11 +1022,41 @@ def testDepthwiseConv2DFilterGradExplicitCompare(self):
                                   padding, "float64")
 
 
-# Please refer to the following gist for more info:
-# https://gist.github.com/duncanriach/4c18cb07a73510c5fcb2deb52adbffaa
-class DepthwiseConv2DDeterministicTest(test.TestCase):
+class DepthwiseConv2DDeterministicTest(DepthwiseConv2DTest):
   """Test determinism-related functionality of tf.nn.depthwise_conv2d."""
 
+  @classmethod
+  def setUpClass(cls):
+    super(DepthwiseConv2DDeterministicTest, cls).setUpClass()
+    # The op-determinism setting can be enabled and disabled on-the-fly.
+    # However, if cuDNN convolution is used (as it is for these tests) then its
+    # setting at the time will influence which algorithm for a particular layer
+    # configuration is cached (independently for XLA and non-XLA operation).
+    #
+    # If DepthwiseConv2DTest were to run first and cause a nondeterministic
+    # cuDNN algorithm to be cached for the configurations used by the
+    # determinism tests below, then it should cause them to fail.
+    #
+    # This persistence of state between tests could also reduce the quality of
+    # the testing for the cases where cuDNN convolution is used for both the
+    # nondeterministic and deterministic operation because algorithm selection,
+    # when this test class and its parent are run (in either order) on an
+    # instance of TensorFlow, may be different from algorithm selection when the
+    # tests were run independently.
+    #
+    # If necessary, a workaround for this issue would be to move
+    # DepthwiseConv2DDeterministicTest into a sparate test file and thereby run
+    # it under a different test.main().
+    #
+    # TODO(duncanriach): Implement cuDNN auto-tuning cache invalidation and
+    # and execute when op-determinism setting is changed.
+    tf_config.enable_op_determinism()
```

Comment:
Thanks for the detailed comment! I'm worried this issue might start to occur if more tests are added or if somehow the test order changes, and this would be hard to debug for someone who didn't read this comment, so could you move this to a separate file in this PR?

Reply:
So grateful that you appreciate my comment, Reed. I will move it into a separate file.